trin-core/src/portalnet/find/iterators/findcontent.rs

```diff
@@ -0,0 +1,715 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+// This basis of this file has been taken from the rust-libp2p codebase:
+// https://github.com/libp2p/rust-libp2p
+
+use std::collections::btree_map::{BTreeMap, Entry};
+use std::time::Instant;
+
+use discv5::kbucket::{Distance, Key};
+
+use super::super::query_pool::QueryState;
+use super::query::{Query, QueryConfig, QueryPeer, QueryPeerState, QueryProgress};
+
+pub enum FindContentResponse<TNodeId> {
+    ClosestNodes(Vec<TNodeId>),
+    Content(Vec<u8>),
+}
```

Comment:
We don't seem to handle `connection_id` `Content` responses? Is there a plan for how to incorporate utp?

Reply:
No, because uTP is not relevant to the query state machine. The query state machine only needs to know whether it holds the desired content or not. Content transmitted via uTP for some query would result in a call to `on_success` with that content.

Please let me know if that does not make sense. I can elaborate further.